src/Tribe/Settings.php

```diff
@@ -328,30 +327,153 @@ public function init_tabs() {
+		 * @param string         $admin_page   The admin page ID.
+		 */
 		$this->no_save_tabs = (array) apply_filters( 'tribe_settings_no_save_tabs', [], $admin_page );
 
 
 		if ( is_network_admin() ) {
+			/**
+			 * Filter the default tab for the network settings page.
+			 *
+			 * @since 4.15.0
+			 *
+			 * @param string $default_tab The default tab for the network settings page.
+			 * @param string $admin_page  The admin page ID.
+			 */
 			$this->default_tab = apply_filters( 'tribe_settings_default_tab_network', 'network', $admin_page );
 		} else {
 			$tabs_keys         = array_keys( $this->tabs );
-			$default_tab       = apply_filters( 'tribe_settings_default_tab', 'general', $admin_page );
-			$this->default_tab = in_array( $default_tab, $tabs_keys ) ? $default_tab : $tabs_keys[0];
+			/**
+			 * Filter the default tab for the settings page.
+			 *
+			 * @since 4.15.0
+			 *
+			 * @param string $default_tab The default tab for the settings page.
+			 * @param string $admin_page  The admin page ID.
+			 */
+			$default_tab       = apply_filters( 'tribe_settings_default_tab', 'viewing', $admin_page );
+			$this->default_tab = in_array( $default_tab, $tabs_keys ) ? $default_tab : array_key_first( $tabs_keys );
 		}
 
-		$this->current_tab     = apply_filters( 'tribe_settings_current_tab', tribe_get_request_var( 'tab', $this->default_tab ), $admin_page );
+		/**
+		 * Filter the current tab.
+		 *
+		 * @since 4.15.0
+		 *
+		 * @param string $current_tab The current tab ID.
+		 * @param string $admin_page  The admin page ID.
+		 */
+		$this->current_tab     = $this->get_current_tab();
 		$this->url             = $this->get_tab_url( $this->current_tab );
+
+		/**
+		 * Filter the fields for save.
+		 *
+		 * @since 4.15.0
+		 *
+		 * @param array  $fields_for_save The fields for save.
+		 * @param string $admin_page      The admin page ID.
+		 */
 		$this->fields_for_save = (array) apply_filters( 'tribe_settings_fields', [], $admin_page );
 
 		do_action( 'tribe_settings_after_do_tabs', $admin_page );
 
+		/**
+		 * Filter the fields for the settings page.
+		 *
+		 * @since 4.15.0
+		 *
+		 * @param array  $fields     The fields for the settings page.
+		 * @param string $admin_page The admin page ID.
+		 */
 		$this->fields = (array) apply_filters( 'tribe_settings_fields', [], $admin_page );
 
 		$this->validate();
 	}
 
+	/**
+	 * Get a specific tab by slug.
+	 * If the slug is not found in the parent tabs, it will then search child tabs for it.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $id The tab ID.
+	 *
+	 * @return Tribe__Settings_Tab|null
+	 */
+	public function get_tab( $id ): ?Tribe__Settings_Tab {
+		// Find tab if a parent.
+		$tab_object = $this->tabs[ $id ] ?? null;
+
+		// Find tab if a child tab.
+		if ( empty( $tab_object ) ) {
+			foreach( $this->tabs as $tab ) {
+				if ( $tab->has_children() && isset( $tab->children[ $id ] ) ) {
```

Comment:
I don't love this... I feel like tabs should already have their children added by the time we're here.

Reply:
We're not adding a child, just making sure we find it